core/src/main/java/org/apache/iceberg/BaseRewriteFiles.java

```diff
@@ -97,11 +103,13 @@ public RewriteFiles validateFromSnapshot(long snapshotId) {
     return this;
   }
 
+  @Override
+  public RewriteFiles setSequenceNumber(long sequenceNumber) {
+    this.replaceSequenceNumber = sequenceNumber;
+    return this;
+  }
+
   @Override
   protected void validate(TableMetadata base) {
-    if (replacedDataFiles.size() > 0) {
-      // if there are replaced data files, there cannot be any new row-level deletes for those data files
-      validateNoNewDeletesForDataFiles(base, startingSnapshotId, replacedDataFiles);
-    }
```

Comment:
This should not be removed. There is a valid use case for the operation to check whether there are conflicts instead of re-sequencing data files. If you want, you can add a configuration method to enable/disable the validation.